### ISSUE TITLE
Turret help

### ISSUE THREAD

**Issue by 117StinkBug** [BrettRyland/BDArmory#674]
I've been trying to create a few rocket, gun, laser, and missile turrets but I cannot seem to get them to move. I thought I'd consult this wiki for help but I don't see any page with useful information. I don't know if I'm not looking closely enough for answers or if the info is hidden in plain sight, but I feel like having a dedicated section for turrets would be useful. A video tutorial for creating turrets would be a nice alternative.

**Comment by BrettRyland** [comment]
Yeah, the wiki is known to be rather lacking in several areas. Unfortunately, writing wiki documentation isn't particularly high priority. It is, however, publicly editable, so feel free to improve it.

For a turret to work, the part config should contain a `ModuleTurret` MODULE and the yaw and pitch transform names should match whatever you've set them to in the model (.mu file). E.g., the goalkeeper turret has
```
  MODULE
  {
    name = ModuleTurret
    turretID = 0

    pitchTransformName = gunPitch
    yawTransformName = gunYaw

    pitchSpeedDPS = 80
    yawSpeedDPS = 80

    maxPitch = 85
    minPitch = -15

    yawRange = 360

    smoothRotation = true
    smoothMultiplier = 10

    audioPath = BDArmory/Sounds/hydraulicLoop
    maxAudioPitch = 0.365
    minAudioPitch = 0.255
    maxVolume = 0.55
  }
```
I don't do the modelling side of things, so if the above isn't sufficient to solve your issues, then maybe one of the others can provide more help.

**Comment by 117StinkBug** [comment]
Do I need to define the pitchTransformName and yawTransformName while making turrets?

**Comment by BrettRyland** [comment]
The default names for them are `pitchTransform` and `yawTransform`. Those transforms will need to exist in the model otherwise you'll get warnings in the log along the lines of
```
[WRN 11:08:21.582] [BDArmory.ModuleTurret]: <turret name> has no pitchTransform
[WRN 11:08:21.582] [BDArmory.ModuleTurret]: <turret name> has no yawTransform
```
if the transforms can't be found.

**Comment by 117StinkBug** [comment]
Alright, thanks for the help!

**Comment by 117StinkBug** [comment]
Actually, now that I think of it, would anyone happen to know how the developer of Aviator Arsenal got the ball turret to require a gunner?

**Comment by BillNyeTheIE** [comment]
Give the part crew capacity and then add
`crewserved = true`
to the ModuleWeapon.
